temporalio/client.py

```diff
@@ -456,6 +460,7 @@ async def start_workflow(
         rpc_timeout: Optional[timedelta] = None,
         request_eager_start: bool = False,
         stack_level: int = 2,
+        priority: Optional[temporalio.common.Priority] = None,
```

Comment:
```suggestion
        priority: temporalio.common.Priority = temporalio.common.Priority.default,
```

If priority is never null when returned from the server, arguably it could be more idiomatic to do this, but no strong opinion

Reply:
I dunno, I like the explicit nature of either providing it or not. Plus just on a practical level I think it makes the code that is accepting it simpler to just check `None` and not set it at all.

Reply:
> I like the explicit nature of either providing it or not

Except when it comes to accessing it as a user, where we don't like that. So we say interceptor outbound input has this optional but interceptor inbound does not which loses symmetry a bit I think. Isn't it true that `Priority()` and `None` effectively mean the same thing?

Reply:
But this is almost exactly contradictory with your suggestion to make the key itself optional. Now, either way, the user needs to see if that was set or not. Having it be optional here also means they have the most obvious way to determine if they explicitly set it, or not. The asymmetry is kind of a good thing I think, because on outbound they get to choose, but on inbound they don't, so it expresses that.

Reply:
> But this is almost exactly contradictory with your suggestion to make the key itself optional

Not really. I support bidirectional optional-ality if you will. Basically if it is optional when they provide it it should be optional when they read it. Basically we either use the language construct to represent default-as-unset or we don't, but half of the time default being an empty object and the other half of the time it being `None` was a bit confusing to me while reading. The user only sees this field when they are changing the default, but it is confusing that I can't expect the default on output be the default on input.

Having said that, my opinion is not strong here nor is it blocking, but I think it may affect a couple of other languages that can have smart parameter defaults like .NET and Ruby.